includes/Core/Modules/Modules.php

```diff
@@ -325,6 +325,52 @@ function ( $data ) {
 				return $data;
 			}
 		);
+
+		add_action(
+			'update_option_googlesitekit_dashboard_sharing',
+			function( $old_values, $values ) {
+				if ( is_array( $values ) && is_array( $old_values ) ) {
+					foreach ( $values as $module_slug => $settings ) {
+						$module = $this->get_module( $module_slug );
+
+						if ( ! $module instanceof Module_With_Service_Entity ) {
+							$changed_settings = false;
+
+							if ( is_array( $settings ) ) {
+								foreach ( $settings as $setting_key => $setting_value ) {
+									if ( is_array( $setting_value ) ) {
+										if ( isset( $old_values[ $module_slug ][ $setting_key ] ) && is_array( $old_values[ $module_slug ][ $setting_key ] ) ) {
+											$difference = array_diff( $setting_value, $old_values[ $module_slug ][ $setting_key ] );
+
+											if ( count( $difference ) ) {
+												$changed_settings = true;
+												break;
+											}
+										} else {
+											$changed_settings = true;
+											break;
+										}
+									} elseif ( $old_values[ $module_slug ][ $setting_key ] !== $setting_value ) {
+										$changed_settings = true;
+										break;
+									}
```

Comment:
I'm finding the logic here just a bit tough to follow 😅 

I got it eventually, but could you add some comments outlining the conditions to consider things changed, as these nested if/elses are tough to scan at a glance.

Reply:
Sounds good. Can you check again @tofumatt ?